driver/controller.go

```diff
@@ -963,3 +973,46 @@ func validateCapabilities(caps []*csi.VolumeCapability) bool {
 
 	return supported
 }
+
+func (d *Driver) tagVolume(ctx context.Context, vol *godo.Volume) error {
+	for _, tag := range vol.Tags {
+		if tag == d.doVolTag {
+			return nil
+		}
+	}
+
+	tagReq := &godo.TagResourcesRequest{
+		Resources: []godo.Resource{
+			godo.Resource{
+				ID:   vol.ID,
+				Type: godo.VolumeResourceType,
+			},
+		},
+	}
+
+	ctx, cancel := context.WithTimeout(ctx, doAPITimeout)
+	resp, err := d.tags.TagResources(ctx, d.doVolTag, tagReq)
+	cancel()
+
+	if resp != nil && resp.StatusCode == http.StatusNotFound {
+		// If the tag doesn't exist, create it and retry.
+		err = d.createTag(ctx)
+		if err != nil {
+			return err
+		}
+		ctx, cancel = context.WithTimeout(ctx, doAPITimeout)
+		resp, err = d.tags.TagResources(ctx, d.doVolTag, tagReq)
+		cancel()
+	}
```

Comment:
I couldn't understand what this code is meant to do. If we create a tag, why do we check for it's existence again? 
Also seems like we're not checking `resp` here and just continue. Finally, we should return from the error check immediately and not continue. I know you're overriding the `err` here, but I feel like that makes the code open for future bugs.

Reply:
When using the TagsService directly (as opposed to as an argument to `CreateVolume`) the tag must exist before you can assign resources (like a volume) to it.  We get a 404 if the tag doesn't exist.  In that case we create the tag, and then retry the `TagResources` which applies the tag to our volume.